File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.java
class Solution {
    public int numSteps(String s) {
        int n = s.length();
        int ans = n - 1;
        int carry = 0;

        for (int i = n - 1; i > 0; i--) {
            int bit = (s.charAt(i) - '0') + carry;

            if (bit == 1) {
                ans++;
                carry = 1;
            }
        }

        return ans + carry;
    }
}

File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution:
    def numSteps(self, s: str) -> int:
        n = len(s)
        ans = n - 1
        carry = 0

        for i in range(n - 1, 0, -1):
            bit = int(s[i]) + carry

            if bit == 1:
                ans += 1
                carry = 1

        return ans + carry